topostats/tracing/skeletonize.py

```diff
@@ -2,78 +2,641 @@
+        for point in mask_coordinates:
+            if self._deletePixelSubit1(point):
+                pixels_to_delete.append(point)
+
+        for x, y in pixels_to_delete:
+            self.mask[x, y] = 0
+
+        # Sub-iteration 2 - binary check
+        mask_coordinates = np.argwhere(self.mask == 1).tolist()
+        for point in mask_coordinates:
+            if self._deletePixelSubit2(point):
+                pixels_to_delete.append(point)
+
+        for x, y in pixels_to_delete:
+            self.mask[x, y] = 0
+
+        if len(pixels_to_delete) == 0:
+            self.skeleton_converged = True
+
+    def _deletePixelSubit1(self, point: list):
+        """Function to check whether a single point should be deleted based
+        on both its local binary environment.
+
+        Parameters
+        ----------
+        point: list
+            List of [x, y] coordinate positions
+
+        Returns
+        -------
+        bool
+            Returns T/F depending if the surrounding points have met the criteria
+            of the binary thin a, b returncount, c and d checks below.
+        """
+
+        self.p7, self.p8, self.p9, self.p6, self.p2, self.p5, self.p4, self.p3 = self.getLocalPixelsBinary(
+            self.mask, point[0], point[1]
+        )
+        return (
+            self._binaryThinCheck_a()
+            and self._binaryThinCheck_b_returncount() == 1
+            and self._binaryThinCheck_c()
+            and self._binaryThinCheck_d()
+        )
+
+    def _deletePixelSubit2(self, point):
+        """Function to check whether a single point should be deleted based
+        on both its local binary environment.
+
+        Parameters
+        ----------
+        point: list
+            List of [x, y] coordinate positions
+
+        Returns
+        -------
+        bool
+            Returns T/F depending if the surrounding points have met the criteria
+            of the binary thin a, b returncount, csharp and dsharp checks below.
+        """
+
+        self.p7, self.p8, self.p9, self.p6, self.p2, self.p5, self.p4, self.p3 = self.getLocalPixelsBinary(
+            self.mask, point[0], point[1]
+        )
+        # Add in generic code here to protect high points from being deleted
+        return (
+            self._binaryThinCheck_a()
+            and self._binaryThinCheck_b_returncount() == 1
+            and self._binaryThinCheck_csharp()
+            and self._binaryThinCheck_dsharp()
+        )
+
+    def _binaryThinCheck_a(self):
+        """Checks the surrounding area to see if the point lies on the edge of the grain.
+        Condition A protects the endpoints (which will be > 2)
+
+        Returns
+        -------
+        bool
+            if point lies on edge of graph and isn't an endpoint.
+        """
+        return 2 <= self.p2 + self.p3 + self.p4 + self.p5 + self.p6 + self.p7 + self.p8 + self.p9 <= 6
+
+    def _binaryThinCheck_b_returncount(self):
```

Comment:
As we're trying to adhere to PEP8 [Class names](https://pep8.org/#class-names) should use CamelCase whilst [function names](https://pep8.org/#function-names) should be lower case (and typically snake_case).

Pylint warns us of this issue ([`C0103` - `invalid name`](https://pylint.pycqa.org/en/latest/user_guide/messages/convention/invalid-name.html).

Further "`_b_`" isn't really informative about what the condition is, if we could name this more descriptively that would be informative.